docs/network-iot/run-an-lns/configure-chirpstack.mdx

```diff
@@ -0,0 +1,88 @@
+---
+id: configure-chirpstack
+title: Configure Chirpstack
+pagination_label: Configure Chirpstack
+sidebar_label: Configure Chirpstack
+description: Configure Chirpstack
+image: https://docs.helium.com/img/link-image.png
+slug: /iot/run-an-lns/configure-chirpstack
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl'
+
+<!-- TODO: Description of necessary keys for configuration and what they do: (payer, owner, delegate) -->
+
+Before getting started, make sure you have the following:
+
+- Helium Network Organizationally Unique Identifier (OUI)
+- Helium LoRaWAN devAddr slab
+
+<!-- TODO: what is config service? -->
+
+### Set Up Chirpstack Docker
+
+1. git clone https://github.com/chirpstack/chirpstack-docker.git
+2. Open configuration/chirpstack/chirpstack.toml
+   1. Under `[network]` Change `net_id` to `0x00003C`
+   2. Under `[network]`, add `dev_addr_prefixes` with the devAddr slab assigned to you. First run
+      `./main org get --OUI <your OUI>` to get your OUI's DevAddr constraints. Then run
+      `./main subnet-mask <your OUI's starting DevAddr> <your OUI's ending DevAddr>` to find your
+      subnets. In the chirpstack toml file put in `dev_addr_prefixes` = [`<your subnets>`], which is
+      formatted with the following rule: `starting devAddr`/`32-log2(devAddr slab size)`. Let's say
+      you requested 8 devAddr slab and the starting devAddr is `00000000`, then you would input
+      `dev_addr_prefixes` = ["00000000/29"]
+   3. Under `[gateway]`, add a line `allow_unknown_gateways=true`
+
+### Set Up Supported LoRaWAN Region
+
+You need to deploy a gateway bridge for each LoRaWAN region you plan on supporting on Chirpstack. By
+default, Chirpstack only supports EU868. This tutorial will show you how to add support for US915,
+and you can replicate the same process for other LoRaWAN regions.
+
+Open docker-compose.yml, you will find a container called `chirpstack-gateway-bridge-eu868`. Add the
+following.
+
+```
+chirpstack-gateway-bridge-us915:
+    image: chirpstack/chirpstack-gateway-bridge:4
+    restart: unless-stopped
+    ports:
+      - 1701:1700/udp
+    volumes:
+      - /helium/configuration/chirpstack-gateway-bridge-us915:/etc/chirpstack-gateway-bridge
+    depends_on:
+      - mosquitto
+    logging: *logs
+```
```

Comment:
Please note that in order for packets to be delivered to your chirpstack-gateway-bridge you'll need to expose this port in a way that is directly reachable by the helium packet router